src/crypto/crypto_context.cc

```diff
@@ -222,11 +261,10 @@ X509_STORE* NewRootCertStore() {
   Mutex::ScopedLock cli_lock(node::per_process::cli_options_mutex);
   if (per_process::cli_options->ssl_openssl_cert_store) {
     X509_STORE_set_default_paths(store);
-  } else {
-    for (X509* cert : root_certs_vector) {
-      X509_up_ref(cert);
-      X509_STORE_add_cert(store, cert);
-    }
+  }
+
+  for (X509* cert : root_certs_vector) {
+    X509_STORE_add_cert(store, cert);
```

Comment:
Since you're here:
```suggestion
    CHECK_EQ(1, X509_STORE_add_cert(store, cert));
```
(I suppose node should do that for `X509_STORE_set_default_paths()`, too.)

Aside: the extraneous `X509_up_ref()` calls don't really matter because the cert objects are immortal anyway.

Reply:
Changes committed.